providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -0,0 +1,129 @@
+    private final FlagSyncServiceGrpc.FlagSyncServiceStub serviceStub;
+
+    public GrpcStreamConnector(final FlagdOptions options) {
+        channel = ChannelBuilder.nettyChannel(options);
+        serviceStub = FlagSyncServiceGrpc.newStub(channel);
+    }
+
+    /**
+     * Initialize gRPC stream connector.
+     */
+    public void init() {
+        Thread listener = new Thread(() -> {
+            try {
+                observeEventStream(blockingQueue, shutdown, serviceStub);
+            } catch (InterruptedException e) {
+                log.log(Level.WARNING, "gRPC event stream interrupted, flag configurations are stale", e);
+            }
+        });
+
+        listener.setDaemon(true);
+        listener.start();
+    }
+
+    /**
+     * Get blocking queue to obtain payloads exposed by this connector.
+     */
+    public BlockingQueue<StreamPayload> getStream() {
+        return blockingQueue;
+    }
+
+    /**
+     * Shutdown gRPC stream connector.
+     */
+    public void shutdown() {
+        shutdown.set(true);
+        channel.shutdown();
+    }
+
+    /**
+     * Contains blocking calls, to be used concurrently.
+     */
+    static void observeEventStream(final BlockingQueue<StreamPayload> writeTo,
+                                   final AtomicBoolean shutdown,
+                                   final FlagSyncServiceGrpc.FlagSyncServiceStub serviceStub)
+            throws InterruptedException {
+
+        final BlockingQueue<GrpcResponseModel> blockingQueue = new LinkedBlockingQueue<>(5);
+        int retryDelay = INIT_BACK_OFF;
+
+        while (!shutdown.get()) {
+            final SyncService.SyncFlagsRequest request = SyncService.SyncFlagsRequest.newBuilder().build();
+            serviceStub.syncFlags(request, new GrpcStreamHandler(blockingQueue));
+
+            while (!shutdown.get()) {
+                GrpcResponseModel response = blockingQueue.take();
+                if (response.isComplete()) {
+                    break;
+                }
+
+                if (response.getError() != null) {
+                    log.log(Level.WARNING,
+                            String.format("error from grpc connection, retrying in %dms", retryDelay),
+                            response.getError());
+                    break;
+                }
+
+                final SyncService.SyncFlagsResponse flagsResponse = response.getSyncFlagsResponse();
+                switch (flagsResponse.getState()) {
+                    case SYNC_STATE_ALL:
+                        if (!writeTo.offer(
+                                new StreamPayload(StreamPayloadType.Data, flagsResponse.getFlagConfiguration()))) {
+                            log.log(Level.WARNING, "Stream writing failed");
+                        }
+                        break;
+                    case SYNC_STATE_UNSPECIFIED:
+                    case SYNC_STATE_ADD:
+                    case SYNC_STATE_UPDATE:
+                    case SYNC_STATE_DELETE:
```

Comment:
I believe we should cover this, otherwise single flag changes will not be propagated - maybe we can add a TODO or a log line for the moment

Reply:
@toddbaert and I discussed these states. While these payloads do show benefits, I don't know of any specific implementation that uses these. I think someday we could move to a V2 spec without these states and completely rely on SYNC_ALL (full flag configuration) only.

Reply:
I will resolve this conversation and follow-up if we see the need